File: python/euler2.py
from decorator import  time_and_memory_decorator as tamd

@tamd
def find_answer():
    answer, current, previous = 0, 0, 1
    while(current < 4000000):
        previous, current = current, current+previous
        if(current % 2 == 0):
            answer += current
    return answer


if __name__ == '__main__':
    print('euler2')
    answer, mem, time = find_answer()

    print(answer)
    print(mem)
    print(time)

File: fortran/euler2.f90
program euler2
  integer :: answer, current, previous, temp
  answer = 0
  current = 2
  previous = 1
  temp = 0

  do while (current < 4000000)
     if (MODULO(current,2) == 0) THEN
        answer = answer + current
     end if
     temp = previous
     previous = current
     current = current+temp
  end do

  print *, answer
end program euler2
